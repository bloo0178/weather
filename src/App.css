.App-Wrapper {
  display: flex;
  flex-direction: column; 
  justify-content: center;
  align-items: center;
  padding: 3em;
}

.Card-Wrapper {
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  padding-bottom: 2em;
}

/* ----------------- TITLE ------------------------ */

.App-Wrapper > h1 {
  padding: 1em;
}

/* ----------------- LOCATION --------------------- */

.location-container {
  width: 20em;
}

.location-container > input {
  width: 20em;
}

/* ------------------- FORECAST ------------------- */

.Forecast-Wrapper {
  width: 75vw;
}

.Forecast-Date {
  padding-top: 1.5em;
  font-weight: bold;
}

.Forecast-Detail {

}


