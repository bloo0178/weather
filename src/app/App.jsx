import React, { Component } from "react";
import styles from "./App.module.scss";
import CurrentWeather from "../Components/CurrentWeather/CurrentWeather";
import CurrentWeatherNEW from '../Components/CurrentWeather/CurrentWeatherNEW';
import LocationSearch from "../Components/LocationSearch/LocationSearch";
import Forecast from "../Components/Forecast/Forecast";
import Test from '../Components/Test/Test';

class App extends Component {
  constructor(props) {
    super(props);
    this.state = {
      location: "",
      locationName: "DEFAULT",
      units: "imperial", //offer ability to select
      tempUnit: "F"
    };
  }

  getLocation = (location, locationName) => {
    this.setState({
      location: location,
      locationName: locationName
    });
  };

  render() {
    const { location, locationName, units, tempUnit } = this.state;
    if (location === "") {
      return (
        <div className={styles.appWrapper}>
          <LocationSearch getLocation={this.getLocation} />
        </div>
      );
    } else {
      return (
        <div className={styles.appWrapper}>
          <LocationSearch getLocation={this.getLocation} />
          <h1>{locationName}</h1>
          <div>
            <CurrentWeather
              location={location}
              units={units}
              tempUnit={tempUnit}
            />
            <CurrentWeatherNEW
              location={location}
              units={units}
              tempUnit={tempUnit}
            />
          </div>
          <div>
            <Forecast location={location} units={units} tempUnit={tempUnit} />
          </div>
        </div>
      );
    }
  }
}

export default App;
