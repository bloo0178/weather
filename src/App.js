import React, { Component } from 'react';
import './App.css';
import Current from './Components/Current-Weather/Current-Weather'
import LocationSearchInput from './Components/Location';
import Forecast from './Components/Forecast/Forecast';
import Search from './Components/Location2'


class App extends Component {
  constructor(props) {
    super(props)
    this.state = {
      //location: '',
      location: 'zip=06831,us', //will need to set to format for lat/ lng
      locationName: 'DEFAULT',
      units: 'imperial', //offer ability to select
      tempUnit: 'F',
    }
  }

  

  //modify this to pull in zip or city/state - cross against city code???
  getLocation = (location, locationName) => {
    this.setState({
      location: location,
      locationName: locationName
    })
  }



  render() {
    if (this.state.location === '') {
      return (
        <div className='App-Wrapper'>
          <LocationSearchInput getLocation={this.getLocation} />
          <h3>Enter a location.</h3>
        </div>
      )
    } else {
      return (
        <div className='App-Wrapper'>
          {/*} <Location getLocation={this.getLocation} /> */}
          <LocationSearchInput getLocation={this.getLocation} />
          <h1>{this.state.locationName}</h1>
          <div>
            <h3>Current Weather</h3>
            <Current
              location={this.state.location}
              units={this.state.units}
              tempUnit={this.state.tempUnit}
            />
          </div>
          <div>
            <h3>Forecast</h3>
            <Forecast
              location={this.state.location}
              units={this.state.units}
              tempUnit={this.state.tempUnit}
            />
          </div>
        </div>
      );
    }
  }
}

export default App;
