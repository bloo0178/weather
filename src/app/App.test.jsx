import React from "react";
import App from "./App";
import {shallow} from 'enzyme';

describe("<App />", ()  => {
    test("renders correctly", () => {
        let wrapper = shallow(<App />); 
        expect(wrapper).toMatchSnapshot();
    })
}); 