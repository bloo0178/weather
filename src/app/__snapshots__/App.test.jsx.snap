// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> renders correctly 1`] = `
<div
  className="Container"
>
  <img
    alt="logo"
    className="Logo"
    src="stormLogo.svg"
  />
  <Wrapper
    getLocation={[Function]}
  />
</div>
`;
